src/java/org/apache/cassandra/repair/RepairRunnable.java

```diff
@@ -461,7 +462,14 @@ public ListenableFuture apply(List<RepairSessionResult> results)
                 // this node is implicitly a participant in this repair, so a single endpoint is ok here
                 if (!endpoints.isEmpty())
                 {
-                    filtered.add(new CommonRange(endpoints, transEndpoints, commonRange.ranges));
+                    Set<InetAddressAndPort> skippedReplicas = Sets.union(Sets.difference(commonRange.endpoints, endpoints),
+                                                                         Sets.difference(commonRange.transEndpoints, transEndpoints));
+                    skippedReplicas.forEach(endpoint -> logger.info("Removing a dead node {} from Repair for ranges {} due to -force", endpoint, commonRange.ranges));
+                    filtered.add(new CommonRange(endpoints, transEndpoints, commonRange.ranges, !skippedReplicas.isEmpty()));
+                }
+                else
+                {
+                    logger.warn("Unable to force repair for {}, as no neighbor nodes are live", commonRange.ranges);
```

Comment:
When an operator reads this warning in the logs, it seems like they might have trouble figuring out for certain what keyspace/tables the repair affects.